### DISCUSSION TITLE
Targeting canvas from another plugin with webgazer

### DISCUSSION THREAD

**Discussion by IvaTomevska** [jspsych/jsPsych#2304] [Q&A]
Hello everyone! I am trying to get webgazer to target a canvas from another plugin in the .html file where I am creating he trials and using webgazer. This is what I mean:
#### plugin.js file has 
(inside the )
```
plugin.trial = function(display_element, trial) {
  ...
         var canvas = document.createElement('canvas');
         canvas.id = 'canvasID';
         document.body.appendChild(canvas);
  ...
}
```

#### index.html file has
```
var trial1 = {
        type: "plugin",
        extensions: [
            {type: 'webgazer', params: {targets: ['#canvasID']}}]}
```

But this is not working. Do you guys know how I can properly call the canvas to target it in the `targets` part when I call webgazer in the trial? Thank you!

**Comment by jodeleeuw** [comment]
It looks like you are appending the `canvas` to the `body`. Have you tried adding it to the `display_element` instead?

```js
display_element.appendChild(canvas);
```

This adds the canvas to the DOM element that jsPsych controls. It may be that the webgazer extension is only looking for targets that are inside the scope of the jsPsych DOM element.

**Comment by jodeleeuw** [comment]
(Converting this to a discussion.)
